sauceplugin-agent/src/main/java/com/saucelabs/teamcity/SauceLifeCycleAdapter.java

```diff
@@ -156,7 +156,7 @@ public void println(String x) {
                     Boolean.TRUE,
                     null
             );
-        } catch (Throwable e) {
+        } catch (AbstractSauceTunnelManager.SauceConnectException e) {
```

Comment:
This very significantly narrows the error scope, also

```
  Process openConnection(
      String username,
      String apiKey,
      File sauceConnectJar,
      String options,
      PrintStream printStream,
      Boolean verboseLogging,
      String sauceConnectPath)
      throws IOException;
```

https://github.com/saucelabs/ci-sauce/blob/6fbf1088e3cc8dafb8092403e715efa4f59398bb/src/main/java/com/saucelabs/ci/sauceconnect/SauceTunnelManager.java#L69

Reply:
This reverts a change that was mistakenly included in my previous commit as part of the following PR: https://github.com/saucelabs/teamcity-sauce-ondemand-plugin/pull/25

I can leave it as it is, but I'm not sure if that's a good idea.

btw. SauceConnectException extends IOException ([see](https://github.com/saucelabs/ci-sauce/blob/6fbf1088e3cc8dafb8092403e715efa4f59398bb/src/main/java/com/saucelabs/ci/sauceconnect/AbstractSauceTunnelManager.java#L710))

Reply:
Maybe revert it in a separate commit then.

Reply:
Also, it's kind of strange that the abstract definition throws SauceConnectException whereas concrete impl throws IOException.